scripts/generator/graphql_generator/csharp/SDK/iOS/CartApplePayEventReceiver.cs.erb

```diff
@@ -0,0 +1,112 @@
+#if UNITY_IPHONE
+namespace <%= namespace %> {
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using <%= namespace %>.SDK;
+    using <%= namespace %>.SDK.iOS;
+    using <%= namespace %>.MiniJSON;
+
+    public partial class Cart : IApplePayEventReceiver {
+
+        public void UpdateSummaryItemsForShippingIdentifier(string serializedMessage) {
+            var message = NativeMessage.CreateFromJSON(serializedMessage);
+
+            SetShippingLine(message.Content, (httpError, graphqlError) => {
+                 if (httpError == null && graphqlError == null) {
+                     var summaryItems = GetSummaryItems();
+                     message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Success, summaryItems).ToJsonString());
+                 } else {
+                     message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Failure).ToJsonString());
+                 }
+            });
+        }
+
+        public void UpdateSummaryItemsForShippingContact(string serializedMessage) {
+
+            var message = NativeMessage.CreateFromJSON(serializedMessage);
+            var contentDictionary = (Dictionary<string, object>)Json.Deserialize(message.Content);
+            var mailingAddressInput = new MailingAddressInput(contentDictionary);
+
+            List<SummaryItem> summaryItems = null;
+            List<ShippingMethod> shippingMethods = null;
+            var updateShippingAddressSuccess = false;
+            var updateEmailSuccess = false;
+
+            Action respondWithSuccess = () => {
+                message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Success, summaryItems, shippingMethods).ToJsonString());
+            };
+
+            SetEmail((String)contentDictionary["email"], (httpError, graphqlError) => {
+                if (httpError == null && graphqlError == null) {
+
+                    if (updateShippingAddressSuccess) {
+                        respondWithSuccess();
+                    } else {
+                        updateEmailSuccess = true;
+                    }
+                } else {
+                    message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Failure).ToJsonString());
+                }
+            });
+
+            SetShippingAddress(mailingAddressInput, (httpError, graphqlError) => {
```

Comment:
I also noticed that `summaryItems` is declared above with `null` and is used in `respondWithSuccess`. If `SetEmail` is called before `SetShippingAddress` it will be null but if it's the other way around it will not be null.